sympy/concrete/expr_with_limits.py

```diff
@@ -276,6 +278,9 @@ def _eval_subs(self, old, new):
         change_index : Perform mapping on the sum and product dummy variables
 
         """
+        from sympy.core.function import AppliedUndef, UndefinedFunction
+
+
```

Comment:
No need for the empty lines (yeah I'm nitpicking here).
More seriously, I've been thinking about ways to avoid imports from inside functions, and I'd like to discuss these before commiting to a specific course. OTOH we already *have* imports inside functions, so it's not doing serious harm to have yet another line of code that does this.